common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -87,25 +88,40 @@ public static Geometry boundary(Geometry geometry) {
     }
 
     public static Geometry buffer(Geometry geometry, double radius) {
-        return buffer(geometry, radius, "");
+        return buffer(geometry, radius, false, "");
     }
 
-    public static Geometry buffer(Geometry geometry, double radius, String params) {
-        if (params.isEmpty()) {
-            return BufferOp.bufferOp(geometry, radius);
-        }
+    public static Geometry buffer(Geometry geometry, double radius, boolean useSpheroid) {
+        return buffer(geometry, radius, useSpheroid, "");
+    }
+
+    public static Geometry buffer(Geometry geometry, double radius, boolean useSpheroid, String params) throws IllegalArgumentException {
+        BufferParameters bufferParameters = new BufferParameters();
 
-        BufferParameters bufferParameters = parseBufferParams(params);
+        // Processing parameters
+        if (!params.isEmpty()) {
+            bufferParameters = parseBufferParams(params);
 
-        // convert the sign to the appropriate direction
-        // left - radius should be positive
-        // right - radius should be negative
-        if (bufferParameters.isSingleSided() &&
-                (params.toLowerCase().contains("left") && radius < 0 || params.toLowerCase().contains("right") && radius > 0)) {
+            // convert the sign to the appropriate direction
```

Comment:
The indention is wrong

Reply:
The indentation is because the 'convert sign' logic is placed inside another if condition to process parameters.